File: algorithm-core/src/main/java/dijkstra/Node.java
package dijkstra;

import java.util.ArrayList;
import java.util.List;

public class Node implements Comparable<Node>{
	public static final int NODE_NUMBER = 50;
	private static final int CONNECTED_LENGTH = 20;
	private boolean oIsAlive = true;
	private boolean oIsVisited;
	private final int oId;
	private final Position oPosition;
	private Node oPrev = null;
	private double oCost = 1000;
	private List<Node> oConnectedNodes = null;
	
	private Node(int aId, Position aPosition) {
		oId = aId;
		oPosition = aPosition;
	}
	
	public static List<Node> createNodes(List<Position> aPositions){
		List<Node> nodes = new ArrayList<>();
		for(int i = 0; i < NODE_NUMBER; i++) {
			nodes.add(new Node(i, aPositions.get(i)));
		}
		
		return nodes;
	}
	
	@Override
	public String toString() {
		return "Node[" + Integer.toString(oId) + "]";
	}
	
	public int getId() {
		return oId;
	}
	
	public void setPrev(Node aNode) {
		oPrev = aNode;
	}
	
	public void setCost(double aCost) {
		oCost = aCost;
	}

	public Position getPosition(){
		return oPosition;
	}
	
	public double getCost() {
		return oCost;
	}
	
	public Node getPrev() {
		return oPrev;
	}

	public void setIsAlive(boolean aIsAlive){ oIsAlive = aIsAlive;}

	public boolean isAlive(){return oIsAlive;}
	
	public String showRoot() {
		StringBuilder builder = new StringBuilder();
		builder.append(this);
		if(oPrev == null) {
			builder.append("");
		}else {
			builder.append("<-" + getPrev().showRoot());
		}
		return builder.toString();
	}
	
	public double getDistance(Node aNode) {
		return Math.sqrt((this.oPosition.oX - aNode.oPosition.oX) * (this.oPosition.oX - aNode.oPosition.oX) + (this.oPosition.oY - aNode.oPosition.oY) * (this.oPosition.oY - aNode.oPosition.oY));
	}
	
	public boolean isWithinRange(Node aNode) {
		if(aNode == this) {
			return false;
		}
		
		return getDistance(aNode) <= CONNECTED_LENGTH;
	}
	
	public void setConnectedNodes(List<Node> aConnectedNodes) {
		oConnectedNodes = aConnectedNodes;
	}
	
	public void setMinimumDistanceNode(List<Node> oNodes) {
		Node minDistanceNode = null;
		for(Node node : oNodes) {
			if(minDistanceNode == null) {
				minDistanceNode = node;
			}
			
			if(getDistance(node) < getDistance(minDistanceNode)) {
				minDistanceNode = node;
			}
		}
		oConnectedNodes.add(minDistanceNode);
	}
	
	public int getConnectedNodesNumber() {
		return oConnectedNodes.size();
	}
	
	public List<Node> getConnectedNodes(){
		return oConnectedNodes;
	}
	
	@Override
	public int compareTo(Node node) {
		if(oCost > node.getCost())
			return 1;
		if(oCost < node.getCost())
			return -1;
		return 0;
	}
	
	public void printPositions() {
		System.out.println("(x,y) = (" + oPosition.oX + " , " + oPosition.oY + ")");
	}
}

File: algorithm-py/dijkstra/Node.py
from dijkstra.Position import *
import math

class Node:
    NODE_NUMBER = 100
    INITIAL_COST = 1000
    CONNECTED_LENGTH = 30

    def __init__(self, id, position):
        self.__id = id
        self.__position = position

    @classmethod
    def createNodes(cls):
        positions = Position.createPositions()
        Position.plotPositions(positions)
        nodes = []
        for i in range(Node.NODE_NUMBER):
            nodes.append(Node(i, positions[i]))

        return nodes

    @classmethod
    def createConnection(cls, nodes):
        connection = np.zeros((Node.NODE_NUMBER, Node.NODE_NUMBER), dtype = int)
        for a in nodes:
            for b in nodes:
                if a.getDistance(b) < Node.CONNECTED_LENGTH:
                    connection[a.__id, b.__id] = 1
                    connection[b.__id, b.__id] = 1

        return connection

    def setPrev(self, prev):
        self.__prev = prev

    def getPrev(self):
        return self.__prev

    def setCost(self, cost):
        self.__cost = cost

    def getCost(self):
        return self.__cost

    def setIsAlive(self, isAlive):
        self.__isAlive = isAlive

    def getIsAlive(self):
        return self.__isAlive

    def printNode(self):
        self.__position.printSelf()

    def getDistance(self, node):
        return math.sqrt((self.__position.x - node.__position.x)**2 + (self.__position.y - node.__position.y)**2)


